virion.md

```diff
@@ -1,161 +1,171 @@
+  "require": {
+    "pmmp/pocketmine-mp": "^4.13.0", # RECOMMENDED unless virion does not use PM API
+    "php": "^8.1" # OPTIONAL but RECOMMENDED if pmmp/pocketmine-mp is omitted
+  },
+  "autoload": {
+    "SOFe\\AwaitGenerator\\": "src"
+  },
+  "x-virion": {
+    "antigen": "SOFe\\AwaitGenerator"
+  }
+}
 ```
 
-`${ANTIBODY_PREFIX}` should be the main namespace of the virion user (namespace of main class for plugins, antigen for virions). Remember to escape backslashes.
-
-## Extending virion CLI functionalities
-Virions can add custom commands by adding a file `cli-map.json` at the project root. The JSON file should contain an object mapping command names to execution files.
-
-If `cli-map.json` is present, Poggit's virion builder will also create a `cli-autoload.php` file which files can include. It will register an autoloader for the src directory.
-
-Command names must not end with `.phar`, otherwise they will be interpreted as plugin phars to shade.
+Then release this virion as with a normal composer library &mdash;
+push to GitHub, [submit on Packagist](https://packagist.org/packages/submit),
+then [release on GitHub](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository).
 
-## Limitations
-- Virion users must not expose virion classes in their API. If they have to expose such objects in their API, they should be wrapped in an interface declared by the virion user.
+A significant difference from Version 1.0 is that
+virions depended by other virions are not double-shaded.
+Shading only takes place once for each virion,
+so different libraries from the same plugin can interact with the same interface
+but cannot interact with libraries included from other plugins.
 
-## Appendix
-### Syntactic references
-Basically, a syntactic reference is defined as:
+## Virion user guide
 
-> A `T_NAMESPACE` (`namespace`), `T_USE` (`use`) or `T_NS_SEPARATOR` (`\`) token, or `T_USE` + `T_FUNCTION` (`use function`) or `T_USE` + `T_CONST` (`use const`), followed by a consecutive series of `T_STRING` (a non-keyword set of characters, i.e. part of the namespace) and `T_NS_SEPARATOR` (`\`).
+To use a virion, firstt create a composer.json for your project:
 
-(Support for `use function` and `use const` was added in Virion Builder Version 1.1)
-
-In simple words, you should **only** reference a virion class/function/constant's namespace using the following formats:
+```yaml
+{
+  "name": "author/project",
+  "require": {
+    "pmmp/pocketmine-mp": "^4.13.0"
+  },
+  "autoload": {
+    "Author\\Plugin": "src"
+  }
+}
+```
 
-```php
-namespace name\space;
-use name\space\Clazz;
-use function name\space\func;
-use const name\space\CONSTANT;
-\name\space\Clazz::method();
+For each virion, add an entry to the `require` attribute
+just like normal composer libraries:
+
+```diff
+  {
+    "name": "author/project",
+    "require": {
++     "sof3/await-generator": "^3.0.0",
+      "pmmp/pocketmine-mp": "^4.13.0"
```

Comment:
does the virion shader remove this as a dependency or are all of pocketmine's classes and libraries shaded into the plugin?

Reply:
@jasonwynn10 I haven't updated the spec yet, but I plan to utilize the fact that type=project in PM will exclude this dependency.